internal/api/v1/client/client.go

```diff
@@ -203,22 +203,41 @@ func (c *APIClient) HealthCheck(ctx context.Context) (map[string]string, error)
 
 // GetInstances lists instances with optional filtering
 func (c *APIClient) GetInstances(ctx context.Context, opts *models.ListOptions) (infrastructure.ListInstancesResponse, error) {
-	endpoint := routes.GetInstancesURL()
+	q := url.Values{}
 	if opts != nil {
-		q := url.Values{}
+		if opts.IncludeDeleted {
+			q.Set("include_deleted", "true")
+		}
 		if opts.Limit > 0 {
 			q.Set("limit", fmt.Sprintf("%d", opts.Limit))
 		}
 		if opts.Offset > 0 {
 			q.Set("offset", fmt.Sprintf("%d", opts.Offset))
 		}
-		if opts.IncludeDeleted {
-			q.Set("include_deleted", "true")
-		}
-		if len(q) > 0 {
-			endpoint = fmt.Sprintf("%s?%s", endpoint, q.Encode())
+		if opts.Status != nil {
+			status := *opts.Status
+			var statusStr string
+			switch status {
+			case models.InstanceStatusUnknown:
+				statusStr = "unknown"
+			case models.InstanceStatusPending:
+				statusStr = "pending"
+			case models.InstanceStatusProvisioning:
+				statusStr = "provisioning"
+			case models.InstanceStatusReady:
+				statusStr = "ready"
+			case models.InstanceStatusTerminated:
+				statusStr = "terminated"
+			default:
+				return infrastructure.ListInstancesResponse{}, fmt.Errorf("invalid instance status: %d", status)
+			}
+			q.Set("status", statusStr)
 		}
 	}
+	endpoint := routes.GetInstancesURL()
+	if len(q) > 0 {
+		endpoint = fmt.Sprintf("%s?%s", endpoint, q.Encode())
+	}
```

Comment:
to prevent needing this check everywhere, I would add the query params as an input to `GetInstancesURL` that passes it down to `BuildURL` where we do this check.